File: src/Array/cpp/SummaryRanges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> str;
    	int index;

    	for (int i = 0; i < nums.size(); i++) {
    		index = i;
    		while (i+1 < nums.size() && nums[i]+1 == nums[i+1])
    			i++;
    		if (index == i)
    			str.push_back(to_string(nums[index]));
    		else
    			str.push_back(to_string(nums[index]) + "->" + to_string(nums[i]));
    	}
    	return str;
    }
};

File: src/haoxiang/Array/SummaryRanges.py
class Solution:
    def summaryRanges(self,nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        start ,res = 0,[]
        for i in range(1, len(nums)+1):
            if i == len(nums) or nums[i] - nums[i-1] != 1:
                if nums[start]!=nums[i-1]:
                    res.append(str(nums[start])+"->"+str(nums[i-1]))
                else:
                    res.append(str(nums[start]))
                start = i
        print res
        return res
    
Test = Solution()
test = [0,1,2,4,5,7]
Test.summaryRanges(test)